src/Fantomas.Tests/CommentTests.fs

```diff
@@ -795,4 +795,44 @@ let ``comments for enum cases, 572``() =
     | CaseB = 1
     // Comment for CaseC
     | CaseC = 2
+"""
+
+[<Test>]
+let ``comments in multi-pattern case matching should not be removed, 813``() =
+    formatSourceString false """
+let f x =
+    match x with
+    | A // inline comment
+    // line comment
+    | B -> Some()
+    | _ -> None""" config
+    |> prepend newline
+    |> should equal """
+let f x =
+    match x with
+    | A // inline comment
+    // line comment
+    | B -> Some()
+    | _ -> None
+"""
+
+[<Test>]
+let ``block comments in multi-pattern case matching should not be removed``() =
+    formatSourceString false """
+let f x =
+    match x with
+    | A
+    (* multi-line
+       block comment *)
+    | B -> Some()
+    | _ -> None""" config
+    |> prepend newline
+    |> should equal """
+let f x =
+    match x with
+    | A
+    (* multi-line
```

Comment:
@nojaf this one is failing as the new line between the pattern and the block comments gets lost

Reply:
Hmm, maybe we can use sepNlnConsiderTrivia (an existing function that adds a newline unless there some trivia content before in the next expression) instead of sepNlnUnlessLastEventIsNewline.